### PR TITLE
Job CLI Upgrade: including configuration key removal 

### PR DESCRIPTION
**Pull request by chesterxgchen**
Fixes # .
### Description
   In some cases, when user changed the local training class constructor or stats class constructor, the arguments name changes
   we need a way to remove the old argument 

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Quick tests passed locally by running `./runtest.sh`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated.

### CODE REVIEW

File: nvflare/tool/job/config/configer.py [hunk 1]
```diff
@@ -177,33 +186,44 @@ def merge_configs(
...
+                                    key_indices.pop(last_token)
                             else:
-                                indices = key_indices.get(key)
-                                for key_index in indices:
-                                    new_value = _cast_type(key_index, cli_value)
-                                    key_index.value = new_value
-                                    parent_key = key_index.parent_key
-                                    if parent_key and isinstance(parent_key.value, ConfigTree):
-                                        parent_key.value.put(key_index.key, new_value)
+                                if cli_value:
```
Comment (IsaacYangSLA) [inline nvflare/tool/job/config/configer.py:217]: This else is really far away from the if statement.  We should think about how to make it easier to read.

**Review by nvidianz — DISMISSED** [review]

File: nvflare/tool/job/config/configer.py [hunk 2]
```diff
@@ -147,6 +147,15 @@ def convert_to_number(value: str):
         return value
 
 
+def get_last_token(input_string):
+    tokens = input_string.split(".")
```
Comment (yhwen) [inline nvflare/tool/job/config/configer.py:151]: Do you need to check input_string not None?

File: nvflare/tool/job/config/configer.py [hunk 3]
```diff
@@ -314,11 +334,19 @@ def _parse_cli_config(
...
-                    raise ValueError("Invalid config data, expecting key, value pair in the format key=value")
-                conf_key = conf[0:index]
-                conf_value = conf[index + 1 :]
+                if conf.endswith("-"):
+                    conf_key = conf[:-1]
+                    conf_value = None
+                else:
+                    index = conf.find("=")
+                    if index == -1:
+                        raise ValueError("Invalid config data, expecting key, value pair in the format key=value")
```
Comment (IsaacYangSLA) [inline nvflare/tool/job/config/configer.py:343]: Line 337 allows conf violates the form of key=value.

**Review by IsaacYangSLA — APPROVED** [review]
Small questions.  We can address them later.

**Review by IsaacYangSLA — APPROVED** [review]
Thanks.  LGTM.

**Review by nvidianz — APPROVED** [review]

**Review by yhwen — APPROVED** [review]

**Pull request merged**